src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ReviewScheduleEtlController.kt

```diff
@@ -0,0 +1,174 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource
+
+import io.swagger.v3.oas.annotations.Hidden
+import mu.KotlinLogging
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.SentenceType
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.dto.CreateInitialReviewScheduleDto
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service.ReviewService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.LegalStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ActionPlanRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.InductionRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ReviewScheduleRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.PrisonerSearchApiService
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Temporary controller class exposing REST API method that triggers the ETL of ReviewSchedule records
+ */
+@Hidden
+@RestController
+class ReviewScheduleEtlController(
+  private val prisonerSearchApiService: PrisonerSearchApiService,
+  private val inductionRepository: InductionRepository,
+  private val actionPlanRepository: ActionPlanRepository,
+  private val reviewScheduleRepository: ReviewScheduleRepository,
+  private val reviewService: ReviewService,
+) {
+
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize(HAS_EDIT_REVIEWS)
+  @RequestMapping(value = ["/action-plans/review-schedules/etl/{prisonId}"])
+  fun createReviewSchedulesForPrisonersInPrison(@PathVariable("prisonId") prisonId: String): ReviewSchedulesEtlResponse {
```

Comment:
I feel like this could do with a 'dry run' flag set defaulted to true - dry run would do everything except the createReviewSchedule. This would allow us to do some confidence testing just in case there is a mistake in the (quite complicated) logic.

Reply:
That is a cracking idea 👍 
(I could have done with that whilst developing it TBH! 😁 , instead I kept deleting records in the dev DB and running it again! 🤣 )